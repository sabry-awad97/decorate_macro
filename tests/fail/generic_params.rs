use decorate_macro::decorate;

fn test_decorator<F, R>(f: F) -> R
where
    F: FnOnce() -> R,
{
    f()
}

#[decorate(test_decorator)]
fn generic_function<T>(x: T) -> T {
    x
}

fn main() {}
