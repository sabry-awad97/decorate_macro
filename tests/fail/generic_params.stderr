error: Decorated functions cannot have generic parameters
  --> tests/fail/generic_params.rs:11:20
   |
11 | fn generic_function<T>(x: T) -> T {
   |                    ^^^
